public/app/features/scopes/dashboards/ScopesDashboardsService.ts

```diff
@@ -88,59 +90,118 @@ export class ScopesDashboardsService extends ScopesServiceBase<ScopesDashboardsS
-              dashboardTitle: dashboard.status.dashboardTitle,
-              items: [],
-            };
-          }
-
-          target[dashboard.spec.dashboard].items.push(dashboard);
-        });
+  public fetchScopeNavigations = async (forScopeNames: string[]) => {
+    if (isEqual(this.state.forScopeNames, forScopeNames)) {
+      return;
+    }
 
-        return acc;
+    const scopeNavigations = await this.apiClient.fetchScopeNavigations(forScopeNames);
+
+    this.updateState({ scopeNavigations });
+  };
+
+  public groupSuggestedItems = (
+    dashboards: ScopeDashboardBinding[],
+    navigations: ScopeNavigation[]
+  ): SuggestedDashboardsFoldersMap => {
+    const folders: SuggestedDashboardsFoldersMap = {
+      '': {
+        title: '',
+        expanded: true,
+        folders: {},
+        dashboards: {},
+        suggestedNavigations: {},
       },
-      {
-        '': {
-          title: '',
-          expanded: true,
-          folders: {},
-          dashboards: {},
-        },
-      }
-    );
+    };
+
+    // Process dashboards
+    dashboards.forEach((dashboard) => {
+      const rootNode = folders[''];
+      const groups = dashboard.status.groups ?? [];
+
+      groups.forEach((group) => {
+        if (group && !rootNode.folders[group]) {
+          rootNode.folders[group] = {
+            title: group,
+            expanded: false,
+            folders: {},
+            dashboards: {},
+            suggestedNavigations: {},
+          };
+        }
+      });
+
+      const targets =
+        groups.length > 0
+          ? groups.map((group) => (group === '' ? rootNode.dashboards : rootNode.folders[group].dashboards))
+          : [rootNode.dashboards];
+
+      targets.forEach((target) => {
+        if (!target[dashboard.spec.dashboard]) {
+          target[dashboard.spec.dashboard] = {
+            dashboard: dashboard.spec.dashboard,
+            dashboardTitle: dashboard.status.dashboardTitle,
+            items: [],
+          };
+        }
+
+        target[dashboard.spec.dashboard].items.push(dashboard);
+      });
+    });
+
+    // Process navigations
+    navigations.forEach((navigation) => {
+      const rootNode = folders[''];
+      const groups = navigation.status.groups ?? [];
+
+      groups.forEach((group) => {
+        if (group && !rootNode.folders[group]) {
+          rootNode.folders[group] = {
+            title: group,
+            expanded: false,
+            folders: {},
+            dashboards: {},
+            suggestedNavigations: {},
+          };
+        }
+      });
+
+      const targets =
+        groups.length > 0
+          ? groups.map((group) =>
+              group === '' ? rootNode.suggestedNavigations : rootNode.folders[group].suggestedNavigations
+            )
+          : [rootNode.suggestedNavigations];
+
+      targets.forEach((target) => {
+        if (!target[navigation.spec.url]) {
+          target[navigation.spec.url] = {
```

Comment:
Seems like maybe just this mapping is different between dashboards and navigations? We could probably generalize function a bit to do both at the same time.